src/rgw/rgw_admin.cc

```diff
@@ -8166,52 +8166,37 @@ int main(int argc, const char **argv)
     if (opt_cmd == OPT::USER_LIST) {
       metadata_key = "user";
     }
-    void *handle;
-    int max = 1000;
-    int ret = store->meta_list_keys_init(dpp(), metadata_key, marker, &handle);
-    if (ret < 0) {
-      cerr << "ERROR: can't get key: " << cpp_strerror(-ret) << std::endl;
-      return -ret;
-    }
 
-    bool truncated;
+    bool truncated = false;
+    void *handle;
     uint64_t count = 0;
+    std::list<std::string> users;
 
-    if (max_entries_specified) {
-      formatter->open_object_section("result");
+    if (max_entries > 1000) {
+      max_entries = 1000;
     }
```

Comment:
If max_entries is accepted on the user then can is please check if it is lower_bounded to zero or not.
if not then, we can do something like this
```suggestion
      max_entries = std::max(0, std::min(max_entries, 1000));
```

Reply:
I have moved this check to backend store. Logic here is` if max_entries <=0` it will print all entries else it will print number of entries specified.

Reply:
Okay 👍 .. for `max_entries= 0` though, I see `keys=[]` returned in ceph.
```bash
~/ceph/build on  master! ⌚ 2:28:06
$ bin/radosgw-admin user list --max-entries 0
2022-04-04T02:28:10.086-0600 7f73e1896500 -1 WARNING: all dangerous and experimental features are enabled.
2022-04-04T02:28:10.100-0600 7f73e1896500 -1 WARNING: all dangerous and experimental features are enabled.
2022-04-04T02:28:10.100-0600 7f73e1896500 -1 WARNING: all dangerous and experimental features are enabled.
{
    "keys": [],
    "truncated": "true",
    "count": 0,
    "marker": "5:1d16e686:users.uid::56789abcdef0123456789abcdef0123456789abcdef0123456789abcdef01234:head"
}
```

Reply:
Yes, this I will raise with them, ideally 0 should print everything otherwise it has no usecase.

Reply:
Okay, Same is the behaviour for `list-buckets`, `list-objects` & `list-parts` etc. You can raise these too if needed.
Just a small diff is that in these cases for -ve values it outputs 0 keys (meaning it caps at 0 and max-limit).